packages/iris-grid/src/mousehandlers/IrisGridContextMenuHandler.tsx

```diff
@@ -925,6 +926,10 @@ class IrisGridContextMenuHandler extends GridMouseHandler {
           title: 'Delete Selected Rows',
           group: IrisGridContextMenuHandler.GROUP_EDIT,
           order: 50,
+          disabled:
+            !isIrisGridTableModelTemplate(model) ||
+            model.inputTable == null ||
+            model.inputTable.keyColumns.length === 0,
```

Comment:
Should just be able to check `selectedRanges.every(range => model.isEditableRange(range)` here. You can't delete rows that are not editable.

Reply:
Just tried doing that, it fixes disables the option for the code mentioned in the ticket (where there is no key columns). However, it also disables the option for the snippet below, when there are key columns defined: 

`from deephaven import empty_table, input_table
source = empty_table(10).update(
    [
        "Sym = (i % 2 == 0) ? `A` : `B`",
        "Marker = (i % 3 == 2) ? `J` : `K`",
        "X = i",
        "Y = sin(0.1 * X)",
    ]
)
input_source = input_table(init_table=source, key_cols=["X", "Y"])`